webapi/Storage/Repository.cs

```diff
@@ -52,7 +52,10 @@ public async Task<bool> TryFindByIdAsync(string id, string? partition = null, Ac
     {
         try
         {
-            callback?.Invoke(await this.FindByIdAsync(id, partition ?? id));
+            T? found = await this.FindByIdAsync(id, partition ?? id);
+
+            callback?.Invoke(found);
```

Comment:
why does saving the call in a variable change the logic?

Reply:
Because if callback was null, Invoke() wouldn't be called and, thus, FindByAsync() neither